azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualNetwork.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import static com.google.common.collect.ImmutableList.copyOf;
+
+import java.util.Map;
+import java.util.List;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class VirtualNetwork {
+
+    @AutoValue
+    public abstract static class AddressSpace{
+
+        public abstract List<String> addressPrefixes();
+
+        @SerializedNames({"addressPrefixes"})
+        public static AddressSpace create(final List<String> addressPrefixes) {
+            return new AutoValue_VirtualNetwork_AddressSpace(copyOf(addressPrefixes));
+        }
+    }
+
+    @AutoValue
+    public abstract static class VirtualNetworkProperties{
+
+        @Nullable
+        public abstract String provisioningState();
+
+        @Nullable
+        public abstract String resourceGuid();
+
+        public abstract AddressSpace addressSpace();
+
+        @Nullable
```

Comment:
@nacx Yes `subnets` here needs to be Nullable as it is an optional property for the create operation and can be omitted. Please refer to [this](https://msdn.microsoft.com/en-us/library/azure/mt163661.aspx).